spec.bs

```diff
@@ -1526,15 +1530,25 @@ The <dfn attribute for="NavigationDestination">sameDocument</dfn> getter steps a
 </div>
 
 <div algorithm>
-  To <dfn>potentially perform scroll restoration</dfn> given a {{Navigation}} object |navigation| and an {{NavigateEvent}} |event|:
+  To <dfn>potentially process scroll behavior</dfn> given a {{NavigateEvent}} |event|:
 
   1. If |event|'s [=NavigateEvent/was intercepted=] is false, then return.
-  1. If |event|'s {{NavigateEvent/navigationType}} was not initialized to "{{NavigationType/traverse}}", then return.
-  1. If |event|'s [=NavigateEvent/scroll restoration behavior=] is "{{NavigationScrollRestoration/manual}}", then return.
-     <p class="note">If it was left as null, then we treat that as "{{NavigationScrollRestoration/after-transition}}", and continue onward.
-  1. If |event|'s [=NavigateEvent/did process scroll restoration=] is true, then return.
-  1. Set |event|'s [=NavigateEvent/did process scroll restoration=] to true.
-  1. [=Restore scroll position data=] given |navigation|'s [=Navigation/current entry=]'s [=NavigationHistoryEntry/session history entry=].
+  1. If |event|'s [=NavigateEvent/scroll behavior=] is "{{NavigationScrollBehavior/manual}}", then return.
+     <p class="note">If it was left as null, then we treat that as "{{NavigationScrollBehavior/after-transition}}", and continue onward.
+  1. If |event|'s [=NavigateEvent/did process scroll behavior=] is true, then return.
+  1. [=Definitely process scroll behavior=] given |event|.
+</div>
+
+<div algorithm>
+   To <dfn>definitely process scroll behavior</dfn> given a {{NavigateEvent}} |event|:
+
+  1. Set |event|'s [=NavigateEvent/did process scroll behavior=] to true.
+  1. If |event|'s {{NavigateEvent/navigationType}} was initialized to "{{NavigationType/traverse}}" or "{{NavigationType/reload}}", then [=restore scroll position data=] given |event|'s [=relevant global object=]'s [=Window/navigable=]'s [=navigable/active session history entry=].
+  1. Otherwise, [=scroll to the fragment=] given |event|'s [=relevant global object=]'s [=associated Document=].
```

Comment:
Should there be an "if" clause in this line? I was surprised to see multiple "otherwise" steps here. Or is the way the spec usually formats >2 branches?

Reply:
This step is just leftover and needs to be deleted. Thanks for catching!